### DISCUSSION TITLE
Regarding the issue of draft entries caused by animation duration

### DISCUSSION THREAD

**Discussion by oeyoews** [TiddlyWiki/TiddlyWiki5#8341] [Q&A]
The destruction timing of the editor in TiddlyWiki seems to be influenced by the animation duration (default is 400ms). If you use ctrl+enter to save a newly created tiddler, and if the user quickly presses some keyboard characters just before the animation ends, the editor hasn't started its destruction yet, and an additional draft entry will be created. Below is an example of this issue, which occurs not only with CodeMirror but also with a regular textarea. Therefore, this bug can be 100% reproduced on tiddlywiki.com (you can also adjust the animation duration from 400ms to 800ms to better reproduce it).

To avoid this rare situation, currently, the best solution is to change the animation duration from 400ms to 0ms.

I hope I have expressed this clearly. @Jermolene  Do you have any good solutions for this issue?

https://github.com/Jermolene/TiddlyWiki5/assets/72405338/227d0609-c10e-4a22-9d32-69c3b4c8ffb8

![image](https://github.com/Jermolene/TiddlyWiki5/assets/72405338/7c361cc9-3af1-4631-aaa5-89f67775cfda)

**Comment by pmario** [comment]
Can you reproduce this problem with tiddlywiki.com? -- I did set the time to 4000 and had no problem there. 
No problem with https://tiddlywiki.com/prerelease/ either. 

Which 3rd party plugins do you use?

**Reply by oeyoews** [reply]
1. 400 --> 1000
2. create new tiddler, and save it use `ctrl+enter`(fastly)
3. press some keys, such <kbd>gggghhhhooo</kbd>

**Comment by Jermolene** [comment]
Thanks @oeyoews – I can confirm that this is a problem (@pmario you need to focus the tiddler body, use the keyboard shortcut to close the tiddler, and then be typing some characters before the animation finishes).

One possibility might be to clone the DOM nodes, and then animate the clone and remove the original DOM nodes immediately.

**Reply by pmario** [reply]
I cannot reproduce the problem with FireFox, nor with Edge on Windows 11 -- strange

**Reply by Jermolene** [reply]
Hi @pmario the steps required to reproduce the problem are very specific – you need to focus the tiddler body and use the keyboard shortcut to close it.

**Comment by Jermolene** [comment]
Hi @oeyoews I've made a PR using the idea of cloning the DOM nodes and it seems to fix the problem – see #8344
